docs.go

```diff
@@ -146,3 +147,35 @@ func flagDetails(flag DocGenerationFlag) string {
 	}
 	return ": " + description
 }
+
+func prepareUsageText(command *Command) string {
+	usageText := ""
+	if command.UsageText != "" {
+		// Remove leading and trailing newlines
+		preparedUsageText := strings.TrimSuffix(command.UsageText, "\n")
+		preparedUsageText = strings.TrimPrefix(preparedUsageText, "\n")
+
+		if strings.Contains(preparedUsageText, "\n") {
+			// Format multi-line string as a code block
+			usageText = fmt.Sprintf("```\n%s\n```\n", preparedUsageText)
```

Comment:
A usage text that already contains a markdown code block will break here. Imagine something like this:

    See code block below:

    ```
    func() { ... }
    ```

    More text here ...


That would be rendered as follows, placing the prefix and suffix in code blocks, and having the inner code block be rendered as regular text.

    ```
    See code block below:

    ```
    func() { ... }
    ```

    More text here ...
    ```


Luckily the markdown spec provides an alternative—text indented to 4 spaces is considered part of a code block. That should allow us to safely turn an entire section into a block of code without worrying about special characters

Reply:
Awesome catch. Updated with the 4 space alternative and it looks to be working well. I am wrapping up tests and should push soon.